packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -446,6 +473,10 @@ export class RunningSummarizer implements IDisposable {
                     });
                     await delay(delaySeconds * 1000);
                 }
+
+                // Make sure the refresh Summary Ack is not being executed.
+                await this.refreshSummaryAckLock;
```

Comment:
Maybe put a comment that because we are pausing op processing, we should not see a new ack processing starting, so waiting for the currently in process should be enough.

I'll point out that this is not exactly true. We pause op processing inside of ContainerRuntime.submitSummary(), i.e. at this like it's still unpaused. So, any await from here to the moment where we pause ops will cause the bug you are trying to fix. The chance of that is extremely low, but it exists. 

It feels like that maybe better solution here is for this await to happen inside of ContainerRuntime.submitSummary(), right where we may call refreshLatestSummaryAckFromServer()? As long as it's not ugly, I'd probably do it. The flows (when we refresh or not ack as part of summary) become unified, and we pause ops right after that.

Reply:
I see. I was assuming the calls to this.generator.summarize   and the submitcallback were close/safe enough. 

Would it be ugly if I pass the refreshSummaryAckLock  as a callback to the summarygenerator and before invoking the submitcallback I would simply wait for it? Once we call submitSummary we almost immediately pause the deltamanager. 
  Something in the lines of: 
         if (refreshLatestAck) {
                await this.waitForRefreshSummaryLockCallback();
            }
            summaryData = await this.submitSummaryCallback({
                fullTree,
                refreshLatestAck,
                summaryLogger: logger,
                cancellationToken,
            });

Reply:
That's probably fine, though I assume callback will be called by ContainerRunime (you reference to submitSummaryCallback suggests it's not).

I think better workflow is one of the following:
- pass context object around through summary code. Make refreshLatestAck part of this context. And then ContainerRUntime.submitSummary() to always call refreshLatestSummaryAckFromServer() which takes this context and decides if it simply waits for existing refresh, or actually does refresh. I.e. remove the need for ConainerRuntime to even know about these details
   - This is a bit hard, because refreshLatestSummaryAckFromServer (and functionality it uses) is on ContainerRuntime. I think it should not be there (ideally all this code is refactored out into reusable layer, such that I can build my own version of ContainerRuntime and reuse this layer, if I want to - right now it's a mix).
- Move explicitly all refresh code out of ContainerRuntime into RunningSummarizer class. And thus allow RunningSummarizer class to directly make these calls and awaits right here. That way you can actually have single lock and not care where ops are paused, as once you took a lock in RunningSummarizer, you disallow any async processing that might happen.